Source/NSTimeZone.m

```diff
@@ -1452,10 +1452,12 @@ + (NSTimeZone*) systemTimeZone
 #endif
 
         localZoneSource = @"function: 'GetTimeZoneInformation()'";
-        if (dst == TIME_ZONE_ID_DAYLIGHT)
-          tzName = tz.DaylightName;
-        else
-          tzName = tz.StandardName;
+        // Only tz.StandardName time zone conversions are supported, as
+        // the Zone-Tzid table lacks all daylight time conversions:
+        // e.g. 'W. Europe Daylight Time' <-> 'Europe/Berlin' is not listed.
+        //
+        // See: https://unicode-org.github.io/cldr-staging/charts/latest/supplemental/zone_tzid.html
+        tzName = tz.StandardName;
```

Comment:
I think this should be conditional for MSVC only. As you mentioned in another PR, MinGW uses IANA timezone names already, and all tests are passing, so I don’t think we should change anything for MinGW: